src/ArduinoJson/MsgPack/BinaryValue.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+ARDUINOJSON_BEGIN_PUBLIC_NAMESPACE
+
+template <typename T>
+class BinaryValue {
+ public:
+  explicit BinaryValue(const T& c) : container_(c) {}
+  operator const T&() const {
+    return container_;
+  }
+
+  typename T::const_iterator begin() const {
+    return container_.cbegin();
+  }
+
+  typename T::const_iterator end() const {
+    return container_.cend();
+  }
+
+  size_t size_bytes() const {
+    return sizeof(typename T::value_type) * container_.size();
+  }
+
+ private:
+  const T& container_;
+};
+
+template <typename T>
+class BinaryValue<T*> {
+ public:
+  explicit BinaryValue(const T* c, size_t size) : data_(c), size_(size) {}
+  operator const T*() const {
+    return data_;
+  }
+
+  const T* begin() const {
+    return data_;
+  }
+
+  const T* end() const {
+    return data_ + size_;
+  }
+
+  size_t size_bytes() const {
```

Comment:
Please rename `size_bytes()` to `size` to be consistent with the rest of the library.